Build scripts do not expand environment variables

When I wrote the instructions on https://sherpa.readthedocs.io/en/4.14.0/developer/index.html#install-from-source-in-conda sherpa used to build with settings like this
`fftw-include_dirs=${ASCDS_LIB}/../include` in sherpa's `setup.cfg`
but when I tried the same thing right now, I get 
```
sherpa/utils/src/tcd/tcdTransform.c:46:19: fatal error: fftw3.h: No such file or directory
     #include "fftw3.h"
                       ^
    compilation terminated.
    error: Command "gcc -pthread -B /nfs/melkor/d1/guenther/soft/mambaforge/envs/sherpaciao/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Isherpa/include -Isherpa/utils/src -Isherpa/utils/src/tcd -I${ASCDS_LIB}/../include -I/tmp/pip-build-env-kfwhz2po/overlay/lib/python3.8/site-packages/numpy/core/include -I/nfs/melkor/d1/guenther/soft/mambaforge/envs/sherpaciao/include/python3.8 -I/nfs/melkor/d1/guenther/soft/mambaforge/envs/sherpaciao/include/python3.8 -c sherpa/utils/src/tcd/tcdTransform.c -o build/temp.linux-x86_64-3.8/sherpa/utils/src/tcd/tcdTransform.o -MMD -MF build/temp.linux-x86_64-3.8/sherpa/utils/src/tcd/tcdTransform.o.d" failed with exit status 1
```
although the variable is set correctly:
```
(sherpaciao) melkor:/melkor/d1/guenther/soft/python/sherpa> ls ${ASCDS_LIB}/../include/fft*
/nfs/melkor/d1/guenther/soft/mambaforge/envs/sherpaciao/lib/../include/fftw3.f
/nfs/melkor/d1/guenther/soft/mambaforge/envs/sherpaciao/lib/../include/fftw3.f03
/nfs/melkor/d1/guenther/soft/mambaforge/envs/sherpaciao/lib/../include/fftw3.h
/nfs/melkor/d1/guenther/soft/mambaforge/envs/sherpaciao/lib/../include/fftw3l.f03
/nfs/melkor/d1/guenther/soft/mambaforge/envs/sherpaciao/lib/../include/fftw3q.f03
```
It works well if I spell out the locations of the files in sherpa's `setup.cfg`
```
fftw-include_dirs=/nfs/melkor/d1/guenther/soft/mambaforge/envs/sherpaciao/lib/../include
```

However, a number of things have changed that are outside of shepa (I'm using mamaba instead of conda, I'm on a different computer, ...) so I don't know what caused this change.

**At this point, I'm not asking for help to debug this, I'm just opening an issue so that I don't forget to check back; this is the type of problem that might not be reproducable and it also outside of the scope of sherpa itself. What we can do is to update https://sherpa.readthedocs.io/en/4.14.0/developer/index.html#install-from-source-in-conda to account for this.**

--- a/docs/developer/index.rst
+++ b/docs/developer/index.rst
@@ -133,7 +133,14 @@ variable ``ASCDS_LIB`` is set to the include directory, when the conda
 environment is activated. Thus, all that is needed is to disable the
 build of external dependencies and to set directories. The following
 lists the lines in ``setup.cfg`` that need to be modified (adjust
-xspec version as needed)::
+xspec version as needed). Note, however, that due to the way that the setup
+and install process works, environment variables need to be expanded by hand.
+So, if your ``ASCDS_LIB`` is set to, e.g.,
+``/data/user/conda/envs/sherpaciao/lib``,
+then you need to expand a line like
+``fftw-include_dirs=${ASCDS_LIB}/../include``
+to ``fftw-include_dirs=/data/user/conda/envs/sherpaciao/lib/../include``
+in the following list::
 
     # GROUP Python module
     disable-group=True

--- a/docs/developer/index.rst
+++ b/docs/developer/index.rst
@@ -133,14 +133,7 @@ variable ``ASCDS_LIB`` is set to the include directory, when the conda
 environment is activated. Thus, all that is needed is to disable the
 build of external dependencies and to set directories. The following
 lists the lines in ``setup.cfg`` that need to be modified (adjust
-xspec version as needed). Note, however, that due to the way that the setup
-and install process works, environment variables need to be expanded by hand.
-So, if your ``ASCDS_LIB`` is set to, e.g.,
-``/data/user/conda/envs/sherpaciao/lib``,
-then you need to expand a line like
-``fftw-include_dirs=${ASCDS_LIB}/../include``
-to ``fftw-include_dirs=/data/user/conda/envs/sherpaciao/lib/../include``
-in the following list::
+xspec version as needed)::
 
     # GROUP Python module
     disable-group=True
@@ -172,6 +165,14 @@ in the following list::
     xspec_version = 12.10.1
     xspec_lib_dirs = ${ASCDS_LIB}
     xspec_include_dirs = ${ASCDS_LIB}/../include
+
+.. note::
+   Due to the way that the setup and install process works, environment
+   variables need to be expanded by hand. So, if your ``ASCDS_LIB`` is set to
+   ``/data/user/conda/envs/sherpaciao/lib`` then you need to expand a line like
+   ``fftw-include_dirs=${ASCDS_LIB}/../include``
+   to ``fftw-include_dirs=/data/user/conda/envs/sherpaciao/lib/../include``
+   in the list above.
 
 To avoid accidentially commiting the modified ``setup.cfg`` into git,
 the file can be marked as "assumed unchanged".

